### PR TITLE
feat(headless): new logContextChanged & logExpandToFullUI action added

### PR DESCRIPTION
**Pull request by mmitiche**
## 1. logContextChanged

In the service console inside Salesforce, when a user updates one of the case fields, the insight panel should automatically trigger a new search query and send the following analytic event:

MewcontextEvent2

A new `logContextChanged` dispatch-able  action is added and and loaded with the new `loadInsightSearchAnalyticsActions` function.

## 2. logExpandToFullUI

In the service console inside Salesforce, when a user opens the full search page, the Hosted Insight Panel should send the ExpandToFullUI  custom event.

NewOpenUI2

A new `logExpandFullUI` dispatch-able  action is added and and loaded with the new `loadInsightSearchAnalyticsActions` function.

### CODE REVIEW

File: packages/headless/src/features/insight-search/insight-search-analytics-actions.ts [hunk 1]
```diff
@@ -25,3 +25,38 @@ export const logQueryError = (error: SearchAPIErrorWithStatusCode) =>
...
+  makeInsightAnalyticsAction(
+    'analytics/contextChanged',
+    AnalyticsType.Search,
+    (client, state) => {
+      const meta = {
+        caseId,
+        caseNumber,
+        caseContext: state.insightCaseContext?.caseContext || {},
+      };
+      client.logContextChanged(meta);
```
Comment (lbergeron) [inline packages/headless/src/features/insight-search/insight-search-analytics-actions.ts:39]: Adding some tests just to validate that the right data is passed down to the client would be nice.

**Review by lbergeron — APPROVED** [review]
Adding tests would be nice. For the rest, it looks pretty good.

**Review by btaillon — APPROVED** [review]

File: packages/headless/src/features/insight-search/insight-search-analytics-actions.ts (hunk 1 above)
Comment (mmitiche) [reply packages/headless/src/features/insight-search/insight-search-analytics-actions.ts:39]: tests added👌

File: packages/headless/src/features/insight-search/insight-search-analytics-actions.test.ts [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+Object.defineProperty(CoveoAnalytics, 'CoveoInsightClient', {
+  value: mockCoveoInsightClient,
+});
+
+const exampleSubject = 'example subject';
+const exampleDescription = 'example description';
+
+describe('logContextChanged', () => {
+  it('should log #logContextChanged whith the right payload', async () => {
```
Comment (lbergeron) [inline packages/headless/src/features/insight-search/insight-search-analytics-actions.test.ts:26]:
```suggestion
  it('should log #logContextChanged with the right payload', async () => {
```

File: packages/headless/src/features/insight-search/insight-search-analytics-actions.test.ts [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+    expect(mockLogContextChanged).toBeCalledTimes(1);
+    expect(mockLogContextChanged.mock.calls[0][0]).toStrictEqual(
+      expectedPayload
+    );
+  });
+});
+
+describe('logExpandToFullUI', () => {
+  it('should log #logExpandToFullUI whith the right payload', async () => {
```
Comment (lbergeron) [inline packages/headless/src/features/insight-search/insight-search-analytics-actions.test.ts:59]:
```suggestion
  it('should log #logExpandToFullUI with the right payload', async () => {
```

**Review by lbergeron — COMMENTED** [review]
Found a few typos

**Comment by mmitiche** [comment]
> Found a few typos

Oops, Thank you! I'll fix them in a separate PR.

**Pull request merged**